src/__experimental__/components/radio-button/radio-button.style.js

```diff
@@ -50,12 +49,20 @@ const RadioButtonStyle = styled(CheckboxStyle)`
       r: 5;
     }
 
-    ${checked && css`
-      circle { fill: ${theme.colors.primary}; }
-    `}
+    ${HiddenCheckableInputStyle}:checked + ${StyledCheckableInputSvgWrapper} circle {
+      fill: ${theme.colors.primary};
+    }
 
     ${disabled && css`
-      circle { fill: ${(checked ? theme.disabled.border : theme.disabled.input)}; }
+      circle { fill: ${theme.disabled.input}; }
+
+      ${disabled && css`
+        circle { fill: ${theme.disabled.input}; }
```

Comment:
A very minor comment: For consistency with the rest of the file, maybe lines 57 and 60 should be formatted like this?
```css
circle {
  fill: ${theme.disabled.input};
}
```